content/en/blog/_posts/2024-03-11-diy-create-your-own-cloud-with-kubernetes-part-3/index.md

```diff
@@ -0,0 +1,160 @@
+
+## Cluster API
+
+Cluster API is an extension for Kubernetes that allows the management of Kubernetes clusters as custom resources within another Kubernetes cluster.
+
+The main goal of the Cluster API is to provide a unified interface for describing the basic entities of a Kubernetes cluster and managing their lifecycle. This enables the automation of processes for creating, updating, and deleting clusters, simplifying scaling, and infrastructure management.
+
+Within the context of Cluster API, there are two terms: **management cluster** and **tenant clusters**.
+
+- **Management cluster** is a Kubernetes cluster used to deploy and manage other clusters. This cluster contains all the necessary Cluster API components and is responsible for describing, creating, and updating tenant clusters. Often it is used just for this purpose.
+- **Tenant clusters** are the user clusters or clusters deployed using the Cluster API. Their creation occurs by describing the relevant resources in the management cluster. They are then used for deploying applications and services by end-users.
+
+It's important to understand that physically, tenant clusters do not necessarily have to run on the same infrastructure with the management cluster; more often, they are running elsewhere.
+
+{{< figure src="clusterapi1.svg" caption="A diagram showing interaction of management Kubernetes cluster and tenant Kubernetes clusters using Cluster API" alt="A diagram showing interaction of management Kubernetes cluster and tenant Kubernetes clusters using Cluster API" >}}
+
+For its operation, Cluster API utilizes the concept of providers which are separate controllers responsible for specific components of the cluster being created. Within Cluster API, there are several types of providers. The main ones are:
+
+ - **Infrastructure Provider**, which is responsible for providing the computing infrastructure, such as virtual machines or physical servers.
+ - **Control Plane Provider**, which provides the Kubernetes control plane, namely the components kube-apiserver, kube-scheduler, and kube-controller-manager.
+ - **Bootstrap Provider**, which is used for generating cloud-init configuration for the virtual machines and servers being created.
+
+To get started, you will need to install the Cluster API itself and one provider of each type. You can find a complete list of supported providers in the project's [documentation](https://cluster-api.sigs.k8s.io/reference/providers.html).
+
+For installation, you can use the `clusterctl` utility, or [Cluster API Operator](https://github.com/kubernetes-sigs/cluster-api-operator) as the more declarative method.
+
+## Our Choice
+
+### Infrastructure Provider
+
+We have chosen [KubeVirt](https://github.com/kubernetes-sigs/cluster-api-provider-uratioclusterctlnubevirt) as our Infrastructure Provider because we use KubeVirt and want to run virtual machines for workers directly inside our management cluster.
```

Comment:
```suggestion
I have chosen [KubeVirt](https://github.com/kubernetes-sigs/cluster-api-provider-uratioclusterctlnubevirt) as the Infrastructure Provider because I use KubeVirt and want to run virtual machines for workers directly inside the management cluster.
```

Let's avoid the use of 'We' everywhere; unless it's for targeting a group of authors

Reply:
@Gauravpadam I've reworked it like this:

> To run Kubernetes clusters using KubeVirt, the [KubeVirt Infrastructure Provider](https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt) must be installed.
This enables the deployment of virtual machines for worker nodes in the same management cluster, where the Cluster API operates.